libslide/src/bignum.rs

```diff
@@ -0,0 +1,100 @@
+use std::fmt;
+const DIGIT_COUNT: usize = 10000;
+
+// this probably can go in utils but I put it in here for now
+trait StringUtils {
+    fn substring(&self, start: usize, len: usize) -> Self;
+}
+
+impl StringUtils for String {
+    fn substring(&self, start: usize, len: usize) -> Self {
+        self.chars().skip(start).take(len).collect()
+    }
+}
+
+pub fn num(input: String) -> Bignum {
+    let bg = Bignum::new(input);
+    bg
+}
```

Comment:
It's probably better to just expose `BigNum` and make its struct fields private.

Reply:
gotcha will change!

Reply:
Please fix